src/components/button/button.component.tsx

```diff
@@ -101,21 +193,29 @@ const Button = ({
     );
   }
 
-  const [internalRef, setInternalRef] = useState(null);
+  invariant(
+    !!(children || iconType),
+    "Either prop `iconType` must be defined or this node must have children."
+  );
+  const internalRef = useRef<HTMLButtonElement>(null);
+  useImperativeHandle<HTMLButtonElement | null, HTMLButtonElement | null>(
+    forwardRef,
+    () => internalRef.current
+  );
 
   const buttonType = as || buttonTypeProp;
 
-  if (subtext.length > 0 && size !== "large") {
+  if (subtext && subtext.length > 0 && size !== "large") {
```

Comment:
This could just be the following as `""` is falsy

```suggestion
  if (subtext && size !== "large") {
```

Reply:
Yeah good spot, and this whole block could probably just be another invariant as well actually